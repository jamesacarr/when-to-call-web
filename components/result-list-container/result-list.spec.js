/* eslint-disable camelcase */

import React from 'react';
import { shallow } from 'enzyme';
import ResultList from './result-list';

const results = [
  { id: '123', primary: '1', secondary: '23' },
  { id: '456', primary: '4', secondary: '56' },
  { id: '789', primary: '7', secondary: '89' }
];
const classes = {
  root: 'root'
};
const selectResult = () => {};

describe('<ResultList/>', () => {
  it('renders null when visible false', () => {
    const wrapper = shallow(<ResultList classes={classes} results={results} selectResult={selectResult} loading={false} visible={false}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders null when loading', () => {
    const wrapper = shallow(<ResultList classes={classes} selectResult={selectResult} loading visible/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders results when visible and not loading', () => {
    const wrapper = shallow(<ResultList classes={classes} results={results} selectResult={selectResult} loading={false} visible/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty state when no results', () => {
    const wrapper = shallow(<ResultList classes={classes} results={[]} selectResult={selectResult} loading={false} visible/>);
    expect(wrapper).toMatchSnapshot();
  });
});
