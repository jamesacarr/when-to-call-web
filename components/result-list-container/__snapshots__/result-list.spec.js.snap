// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultList/> renders empty state when no results 1`] = `
<WithStyles(Paper)
  className="root"
>
  <WithStyles(List)>
    <EmptyRow />
  </WithStyles(List)>
</WithStyles(Paper)>
`;

exports[`<ResultList/> renders null when loading 1`] = `""`;

exports[`<ResultList/> renders null when visible false 1`] = `""`;

exports[`<ResultList/> renders results when visible and not loading 1`] = `
<WithStyles(Paper)
  className="root"
>
  <WithStyles(List)>
    <ResultRow
      divider={true}
      key="123"
      onClick={[Function]}
      result={
        Object {
          "id": "123",
          "primary": "1",
          "secondary": "23",
        }
      }
    />
    <ResultRow
      divider={true}
      key="456"
      onClick={[Function]}
      result={
        Object {
          "id": "456",
          "primary": "4",
          "secondary": "56",
        }
      }
    />
    <ResultRow
      divider={false}
      key="789"
      onClick={[Function]}
      result={
        Object {
          "id": "789",
          "primary": "7",
          "secondary": "89",
        }
      }
    />
  </WithStyles(List)>
</WithStyles(Paper)>
`;
